src/lib/gssapi/krb5/accept_sec_context.c

```diff
@@ -424,6 +424,25 @@ kg_process_extension(krb5_context context,
 /* A zero-value channel binding, for comparison */
 static const uint8_t null_cb[CB_MD5_LEN];
 
+static krb5_error_code
+check_ap_options_cbt(krb5_context context, krb5_authdata **authdata,
+                     krb5_boolean *client_asserted_cd)
+{
+    uint32_t ad_ap_options;
+
+    *client_asserted_cd = FALSE;
+
+    if (authdata[1] != NULL || authdata[0]->length != sizeof(uint32_t))
+        return KRB5KRB_AP_ERR_MSG_TYPE;
+
+    memcpy(&ad_ap_options, authdata[0]->contents, sizeof(uint32_t));
```

Comment:
shouldn't this be converted via endian aware macros ?

Reply:
I guess it should, i'll look for examples.

Reply:
You want load_32_le() from k5-platform.h, and store_32_le() for the reverse operation.